core/include/usBundleContext.h

```diff
@@ -691,7 +736,18 @@ class US_Core_EXPORT BundleContext
   friend class BundlePrivate;
   template<class S> friend struct ServiceHolder;
 
+  // allow templated code to use the internal logger
+  template<class S, class TTT, class R> friend class BundleAbstractTracked;
+  template<class S, class T> friend class ServiceTracker;
+  template<class S, class TTT> friend class ServiceTrackerPrivate;
+  template<class S, class TTT> friend class TrackedService;
+
   BundleContext(BundlePrivate* bundle);
+
+  // Not for use by clients of the Framework.
+  // Provides access to the Framework's log sink to allow templated code
+  // to log diagnostic information. 
+  std::shared_ptr<LogSink> GetLogSink();
```

Comment:
What about moving the `GetLogSink()` to BundleContexPrivate? I think we can then also get rid of the new friend declarations and use `GetPrivate()` instead.